pkg/controllers/machineinventoryselector/bootstrap.go

```diff
@@ -76,6 +77,27 @@ func (h *handler) bootstrapReadyHandler(obj *v1beta1.MachineInventorySelector, s
 	// if the bootstrap plan is applied bootstrap is ready
 	if inventory.Status.Plan.AppliedChecksum == status.BootstrapPlanChecksum {
 		v1beta1.BootstrapReadyCondition.SetError(&status, v1beta1.BootstrapReadyReason, nil)
+
+		if inventory.Status.Plan.FailedChecksum == status.BootstrapPlanChecksum {
+			logrus.Warn("boostrap plan failed...")
+			return status, nil
+		}
+
+		logrus.Info("bootstrap plan succeeded, setting plan to stop elemental-system-agent")
+		plan := getStopElementalAgentPlan()
+
+		planSecret, err := h.SecretCache.Get(inventory.Status.Plan.SecretRef.Namespace, inventory.Status.Plan.SecretRef.Name)
+		if err != nil {
+			logrus.Warn("failed to get plan secret, stop elemental-system-agent plan is not set")
+			return status, nil
+		}
+
+		planSecret.Data["plan"] = plan
```

Comment:
Im wondering, does elemental-system-agent and rancher-system-agent use the same secret for plan deploying? Could this override the plan set for the rancher-system-agent if they are using the same plan?

Reply:
machineinventory controller creates a secret, this secret is later picked up by machineinventoryselector controller and populated with a plan.

Reply:
> Im wondering, does elemental-system-agent and rancher-system-agent use the same secret for plan deploying? Could this override the plan set for the rancher-system-agent if they are using the same plan?

They are not using the same plan, that's why we need both. System agent listens only a single plan hence there is no easy way of making a handover from one to the other. We have no or little control over the rancher-system-agent plan and its setup. This is all managed by the bootstrap.sh script.